pkg/scm/git/git.go

```diff
@@ -537,7 +570,66 @@ func (h *stiGit) SubmoduleUpdate(repo string, init, recursive bool) error {
 		Stderr: os.Stderr,
 		Dir:    repo,
 	}
-	return h.runner.RunWithOptions(opts, "git", updateArgs...)
+	return h.RunWithOptions(opts, "git", updateArgs...)
+}
+
+// LsTree returns a slice of os.FileInfo objects populated with the paths and
+// file modes of files known to Git.  This is used on Windows systems where the
+// executable mode metadata is lost on git checkout.
+func (h *stiGit) LsTree(repo, ref string, recursive bool) ([]os.FileInfo, error) {
+	rx := regexp.MustCompile("([0-7]{6}) [^ ]+ [0-9a-f]{40}\t(.*)")
```

Comment:
I'd suggest making this regexp global (i.e., in the package level), so that we don't recompile in every call to LsTree (I see that at least when there are submodules we call this func in a loop/recursively).